code/controllers/subsystem/npcpool.dm

```diff
@@ -21,6 +21,9 @@ SUBSYSTEM_DEF(npcpool)
     while(currentrun.len)
         var/mob/living/simple_animal/SA = currentrun[currentrun.len]
         --currentrun.len
+        if(!SA)
+            GLOB.simple_animals[AI_ON] -= SA
+            continue
```

Comment:
This is a cheap way to fix it and not proper at all. We need to find the *source* of why it is null, and not paper over the problem.

Reply:
Copied from idlenpcpool.dm
But I'll see about finding the source of the issue

Reply:
Tried to find some causes. Inbetween update (else I forget):
All destroys will remove the animal from the AIStatus they are assigned to.
There are some ways that AIStatus is changed without it using toggle_ai() and thus removing it from the list that NPCPool uses.
I'll check those ways to see if they fuck shit up

Reply:
Probably won't have time myself to find the real cause. This will however fix the issue for now since it is affecting the game. Bots not working etc.

Reply:
could you add some kind of logging to these null checks so that we have a lead on what's causing the nulls, since these kinds of errors will have impacts in other parts of the code, too

Reply:
I'll add some debugging with tons of info.
Another reason that comes to mind for it to happen might be due to the resuming of the SS.